manager/controlapi/ca_rotation.go

```diff
@@ -104,171 +103,237 @@ func validateExternalCAURL(dialer *net.Dialer, tlsOpts *tls.Config, caURL string
 	if parsed.Scheme != "https" {
 		return errors.New("invalid HTTP scheme")
 	}
-	conn, err := tls.DialWithDialer(dialer, "tcp", getConnectionStringFromHost(parsed.Host), tlsOpts)
+	host, port, err := net.SplitHostPort(parsed.Host)
+	if err != nil {
+		return err
+	}
+	if port == "" {
```

Comment:
I'm not sure this is reachable - it looks like `SplitHostPort` returns an error in this case. I expected that if `SplitHostPort` returns an error, we would assume the address does not have a port.